src/Microsoft.CodeQuality.Analyzers/Core/Maintainability/ReviewSqlQueriesForSecurityVulnerabilities.cs

```diff
@@ -0,0 +1,206 @@
+                    AnalyzeMethodCall(operationContext, creation.Constructor, symbol, creation.Arguments, creation.Syntax, isInDbCommandConstructor, isInDataAdapterConstructor, callingDataAdapterConstructor);
+                }, OperationKind.ObjectCreation);
+
+                // If an object calls a constructor in a base class or the same class, this will get called.
+                operationBlockStartContext.RegisterOperationAction(operationContext =>
+                {
+                    var invocation = (IInvocationOperation)operationContext.Operation;
+
+                    // We only analyze constructor invocations
+                    if (invocation.TargetMethod.MethodKind != MethodKind.Constructor)
+                    {
+                        return;
+                    }
+
+                    // If we're calling another constructor in the same class from this constructor, assume that all parameters are safe and skip analysis. Parameter usage
+                    // will be analyzed there
+                    if (invocation.TargetMethod.ContainingType == symbol.ContainingType)
+                    {
+                        return;
+                    }
+
+                    var callingDataAdapterConstructor = invocation.TargetMethod.ContainingType.AllInterfaces.Contains(iDataAdapterType);
+                    AnalyzeMethodCall(operationContext, invocation.TargetMethod, symbol, invocation.Arguments, invocation.Syntax, isInDbCommandConstructor, isInDataAdapterConstructor, callingDataAdapterConstructor);
+                }, OperationKind.Invocation);
+
+                operationBlockStartContext.RegisterOperationAction(operationContext =>
+                {
+                    var propertyReference = (IPropertyReferenceOperation)operationContext.Operation;
+
+                    // We're only interested in implementations of IDbCommand.CommandText
+                    if (!propertyReference.Property.IsImplementationOfInterfaceMember(commandTextProperty))
+                    {
+                        return;
+                    }
+
+                    // Make sure we're in assignment statement
+                    if (!(propertyReference.Parent is IAssignmentOperation assignment))
+                    {
+                        return;
+                    }
+
+                    // Only if the property reference is actually the target of the assignment
+                    if (assignment.Target != propertyReference)
+                    {
+                        return;
+                    }
+
+                    ReportDiagnosticIfNecessary(operationContext, assignment.Value, assignment.Syntax, propertyReference.Property, symbol);
+                }, OperationKind.PropertyReference);
+            });
+        }
+
+        private void AnalyzeMethodCall(OperationAnalysisContext operationContext,
```

Comment:
I don't see us bailing out from this method for following cases:

1. `constructorSymbol` doesn't belong to a type that neither implements  `IDbCommand` nor `IDataAdapter`.
2. `isInDbCommandConstructor && constructorSymbol.ContainingType.AllInterfaces.Contains(iDbCommandType) || isInDataAdapterConstructor && callingDataAdapterConstructor)

Reply:
1 is fixed.

2 is taken care of by https://github.com/dotnet/roslyn-analyzers/pull/1504/files#diff-373b4905b697892ac6a1afa3e28f29e2R164.